### PR TITLE
[xpu] [pir] add fc_xpu_fuse_pass

### PR DESCRIPTION
**Pull request by NeroLoh**
### PR Category
Others 

### PR Types
Others 

### Description
pir add fc_xpu_fuse_pass

### CODE REVIEW

File: paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/phi/core/dense_tensor.h"
+#include "paddle/pir/include/pass/pass.h"
+#include "paddle/pir/include/pass/pass_registry.h"
+#include "paddle/pir/include/pattern_rewrite/pattern_match.h"
+
+namespace {
+
+int ConvertActivationType(std::string act_type) {
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc]:
```suggestion
int ConvertActivationType(const std::string& act_type) {
```

File: paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      const auto &add = pat.Op(paddle::dialect::AddOp::name());
+      add({&pat.Tensor("mul_out"), &pat.Tensor("bias")},
+          {&pat.Tensor("add_out")});
+    }
+    pat.AddConstraint([&](const paddle::drr::MatchContext &match_ctx) {
+      auto persist_attr = match_ctx.Tensor("w").attribute<pir::BoolAttribute>(
+          kAttrIsPersistable);
+      if (!(persist_attr && persist_attr.data())) {
+        return false;
+      }
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc]:
```suggestion
      if (!pir::ValueIsPersistable(match_ctx.Tensor("w"))) {
        return false;
      }
```

Comment (NeroLoh) [reply paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc]: done

File: paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc (hunk 1 above)
Comment (NeroLoh) [reply paddle/fluid/pir/transforms/xpu/fc_xpu_fuse_pass.cc]: done

**Pull request closed without merging**